scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -221,17 +222,35 @@ class FossId(
                         .checkResponse("create project")
                 }
 
-                log.info { "Creating scan for URL $url and revision $revision ..." }
+                val scans = service.listScansForProject(user, apiKey, projectCode)
+                    .checkResponse("list scans for project").data
+                requireNotNull(scans)
 
-                val scanCode = createScan(projectCode, url, revision)
-
-                log.info { "Initiating download from Git repository ..." }
+                val existingScan = scans.sortedByDescending { scan -> scan.id }.find { scan ->
+                    scan.gitBranch == revision && scan.gitRepoUrl == url
+                }
 
-                service.downloadFromGit(user, apiKey, scanCode)
-                    .checkResponse("download data from Git", false)
+                val scanCode = kotlin.run {
+                    if (existingScan == null) {
+                        log.info { "No scan found for url=$url and revision=$revision. Creating scan ..." }
+                        val scanCode = createScan(projectCode, url, revision)
+
+                        log.info { "Launching data download ..." }
+                        val responseDownload = service.downloadFromGit(user, apiKey, scanCode)
+                        responseDownload.checkResponse("download data from Git", false)
+                        scanCode
+                    } else {
+                        log.info { "Scan found for url=$url and revision=$revision." }
+
+                        val scanCode = existingScan.code
+                        requireNotNull(scanCode) {
+                            "FossId returned a null scancode for an existing scan"
+                        }
+                        scanCode
```

Comment:
As `requireNotNull` returns a non-null version of its argument, you can omit this line (and then inline `existingScan.code` to `requireNotNull`).